python/kserve/kserve/model_server.py

```diff
@@ -56,8 +57,10 @@
                     help="Output a log per request with latency metrics.")
 parser.add_argument("--configure_logging", default=True, type=lambda x: bool(strtobool(x)),
                     help="Whether to configure KServe and Uvicorn logging")
-parser.add_argument("--log_config_file", default=None, type=str,
-                    help="File path containing UvicornServer's log config. Needs to be a yaml or json file.")
+parser.add_argument("--log_config", default=None, type=str,
```

Comment:
Did you try setting the type to dict so you do not need to parse the json in your code ?

Reply:
@yuzisun I haven't tried since in theory we shouldn't be able to use a file path right? Or will it be possible?

Reply:
I am also trying to test type=dict in argparse but I don't find a good way to pass the dict via command line (the str format seems not ok when type is dict). Does it work for you?

Reply:
you are right, this does not work

Reply:
I don't know if it makes sense to change the name of this arg since it's very difficult to pass a dict as an argument from the command line. If instantiating the model server programmatically it is possible to pass the argument as dictionary. That's why we've named the function parameter `log_config` (as it can can be a file path or a dictionary), while keeping the CLI argument as `log_config_file`. Does this make sense?